File: 0187-repeated-dna-sequences/0187-repeated-dna-sequences.java
class Solution {
    public List<String> findRepeatedDnaSequences(String s) {
        HashSet<String> hashset = new HashSet<>();
        List<String> res = new ArrayList<>();
        
        for(int i = 0; i < s.length() - 9; i++) {
            String str = s.substring(i, i+10);
            if(hashset.contains(str)) {
                if(!res.contains(str)) {
                    res.add(str);
                }
            }
            hashset.add(str);
        }
        
        return res;
        
        
    }
}

File: 0187-repeated-dna-sequences/0187-repeated-dna-sequences.py
class Solution:
    def findRepeatedDnaSequences(self, s: str) -> List[str]:
        hashset = set()
        res = []
        for i in range(len(s) - 9):
            if s[i:i+10] in hashset:
                if s[i:i+10] not in res:
                    res.append(s[i:i+10])
            hashset.add(s[i:i+10])
        return res